scheduler/scheduler_system.go

```diff
@@ -157,6 +157,18 @@ func (s *SystemScheduler) process() (bool, error) {
 	s.stack = NewSystemStack(s.sysbatch, s.ctx)
 	if !s.job.Stopped() {
 		s.stack.SetJob(s.job)
+
+		// Fetch node pool and global scheduler configuration to determine
+		// which values to use.
+		pool, err := s.state.NodePoolByName(ws, s.job.NodePool)
+		if err != nil {
+			return false, fmt.Errorf("failed to get job node pool %q: %v", s.job.NodePool, err)
+		}
+		_, schedConfig, err := s.state.SchedulerConfig()
+		if err != nil {
+			return false, fmt.Errorf("failed to get scheduler configuration: %v", err)
+		}
+		s.stack.SetSchedulerConfiguration(schedConfig.WithNodePool(pool))
```

Comment:
We're not taking the same approach as we did in the generic scheduler because this is the only caller of `SetJob` and `SchedulerConfiguration`, right?

Reply:
That's right, although it may be nice to keep them consistent, so I think I will follow the same pattern here 👍

Reply:
Updated in https://github.com/hashicorp/nomad/pull/17598/commits/742263623d034bf2ca6b3e56bc5f67292200cc6b